app/static/src/app/components/mixins/selectVariables.ts

```diff
@@ -0,0 +1,77 @@
+import { Store } from "vuex";
+import { computed } from "vue";
+import { AppState } from "../../store/appState/state";
+import { GraphsAction } from "../../store/graphs/actions";
+
+export interface SelectVariablesMixin {
+    startDrag: (evt: DragEvent, variable: string) => void;
+    endDrag: () => void;
+    onDrop: (evt: DragEvent) => void;
+    removeVariable: (srcGraphIdx: number, variable: string) => void;
+}
+
+export default (
+    store: Store<AppState>,
+    emit: (event: string, ...args: unknown[]) => void,
+    hiddenVariables: boolean,
+    graphIndex?: number
+): SelectVariablesMixin => {
+    const thisSrcGraph = hiddenVariables ? "hidden" : graphIndex!.toString();
```

Comment:
sorry finding srcGraph naming a bit confusing here... its actually not the graph right? but the box where the variables sit?

Reply:
Yeah, in the context of dropping a variable onto a graph config or hidden variables area it's the corresponding area that it was dragged from. We set that value in the event so that the drop target knows where it came from. Would "srcConfig" or "srcGraphConfig" be better?